### PR TITLE
Update span name recommendation for HTTP Server

### PR DESCRIPTION
**Pull request by gouthamve**
This is to bring it inline with: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-server-semantic-conventions

## Merge requirement checklist

* [ ] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/http/http-spans.md [hunk 1]
```diff
@@ -325,7 +325,7 @@ This span type represents an inbound HTTP request.
 
 For an HTTP server span, `SpanKind` MUST be `Server`.
 
-Given an inbound request for a route (e.g. `"/users/:userID?"`) the `name` attribute of the span SHOULD be set to this route.
+Given an inbound request for a route (e.g. `"/users/:userID?"`) the `name` attribute of the span SHOULD be set to `{http.request.method} {http.route}` (eg. `GET /users/:userID?`).
 
 If the route cannot be determined, the `name` attribute MUST be set as defined in the general semantic conventions for HTTP.
```
[the remark this answers is not in the document]
Comment (joaopgrassi) [reply docs/http/http-spans.md:330]: I'd also think we should remove it. The `HTTP Client` section also does not have this duplication, which is another indication we should probably remove.

[the remark this answers is not in the document]
Comment (gouthamve) [reply docs/http/http-spans.md]: Done

**Review by pyohannes — APPROVED** [review]

**Comment by pyohannes** [comment]
@gouthamve Please rebase to make the checks pass.

**Comment by pellared** [comment]
1. I think the link in the description should be https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#http-server
2. Shouldn't you also update the [examples](https://github.com/open-telemetry/semantic-conventions/blob/911177868ba748a61985affc261c8ad48e5df251/docs/http/http-spans.md#examples)?

**Comment by AlexanderWert** [comment]
@gouthamve

I don't fully get the change here, I guess I'm missing some context here.
Why would we stop recommending to use the `route` in HTTP server span names?

**Comment by pyohannes** [comment]
@AlexanderWert 

> I don't fully get the change here, I guess I'm missing some context here.
> Why would we stop recommending to use the route in HTTP server span names?

It's not about not using the route anymore. How to set the span name is clearly defined in a [dedicated section of this document](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#name):

> HTTP server span names SHOULD be `{method} {http.route}` if there is a (low-cardinality) `http.route` available (see below for the exact definition of the [`{method}`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http spans.md#method-placeholder) placeholder).
>
> If there is no (low-cardinality) `http.route` available, HTTP server span names SHOULD be [{method}](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#method-placeholder).

The paragraph to remove in this PR suggested that the span name should be set just to `{http.route}`, without a trailing `{method}`. That's incorrect.

**Comment by AlexanderWert** [comment]
> @AlexanderWert
> 
> > I don't fully get the change here, I guess I'm missing some context here.
> > Why would we stop recommending to use the route in HTTP server span names?
> 
> It's not about not using the route anymore. How to set the span name is clearly defined in a [dedicated section of this document](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#name):
> 
> > HTTP server span names SHOULD be `{method} {http.route}` if there is a (low-cardinality) `http.route` available (see below for the exact definition of the [`{method}`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http spans.md#method-placeholder) placeholder).
> > If there is no (low-cardinality) `http.route` available, HTTP server span names SHOULD be [{method}](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#method-placeholder).
> 
> The paragraph to remove in this PR suggested that the span name should be set just to `{http.route}`, without a trailing `{method}`. That's incorrect.

@pyohannes Thanks! Now I get it!

**Review by trask — APPROVED** [review]
thx @gouthamve!

and thx @pyohannes for the [explainer](https://github.com/open-telemetry/semantic-conventions/pull/287#issuecomment-1729655698)!

**Comment by trask** [comment]
@open-telemetry/specs-semconv-maintainers I think this is good to merge

**Pull request merged**